lib/engine/config/game/g_1849.rb

```diff
@@ -473,19 +473,120 @@ module G1849
   },
   "phases": [
     {
-      "name": "4",
+      "name": "4H",
       "train_limit": 4,
       "tiles": [
         "yellow"
       ],
       "operating_rounds": 1
+    },
+    {
+      "name": "6H",
+      "train_limit": 4,
+      "tiles": [
+        "yellow",
+        "green"
+      ],
+      "operating_rounds": 2
+    },
+    {
+      "name": "8H",
+      "train_limit": 3,
+      "tiles": [
+        "yellow",
+        "green"
+      ],
+      "operating_rounds": 2
+    },
+    {
+      "name": "10H",
+      "train_limit": 2,
+      "tiles": [
+        "yellow",
+        "green",
+        "brown"
+      ],
+      "operating_rounds": 3
+    },
+    {
+      "name": "12H",
+      "train_limit": 2,
+      "tiles": [
+        "yellow",
+        "green",
+        "brown"
+      ],
+      "operating_rounds": 3
+    },
+    {
+      "name": "16H",
+      "train_limit": 2,
+      "tiles": [
+        "yellow",
+        "green",
+        "brown"
+      ],
+      "operating_rounds": 3
     }
   ],
   "companies": [
   ],
   "corporations": [
   ],
   "trains": [
+    {
+      "name": "4H",
+      "num": 4,
+      "distance": 99,
+      "price": 100,
+      "rusts_on": "8H",
+      "hex_edge_limit": 4
```

Comment:
can we change this to

"distance": { "hex": 4 }

Reply:
+1, I think we should also add the R6H type here somehow

Reply:
"distance": {"hex": 4, "terrain_cost": {"broad": 2}}
or something similar.

Reply:
I considered this but you get a lot of mileage out of having `distance` be a numeric, and a lot more code would have to be changed to support hex-typed `distance` than how I've done it. Also, I'm not sure I see the benefit of allowing players to select their routes hex by hex - ultimately revenue centers are what matter for a train run.

Reply:
which part needs to change? distance is already not numeric in some games

i don't think we're suggesting allowing players to select routes hex by hex

Reply:
Sorry, I should have been more clear. My primary concern is with `compute_stops` which expects that `distance` be mostly about revenue centers, and uses `route.visited_stops` (again, concerned with revenue centers) in its computation. wrt selecting routes hex by hex - of course, no, but that's the ad absurdum result of redefining e.g. `Connection` (used in `visited_stops`) to include hexes.

Reply:
if you change distance to be { 'hex': 4 }  you could  just check distance.is_a?(Numeric) || distance['hex']

if you don't like this perhaps then change 'hex_limit': true and keep distance a real number

Reply:
I took your second suggestion. I don't see a lot of value in specifying this further in config given that the few games that have hex trains and narrow gauge all use them quite differently:

- 18Ireland: Trains can't run on narrow gauge.
- 1858: Has 4 different kinds of train only one of which counts hexes and only one of which uses narrow gauge (they're not the same)
- 1849: As implemented here.

Reply:
yeah probably the case.